lib/config/rules/promise.js

```diff
@@ -1,13 +1,33 @@
 // see https://github.com/xjamundx/eslint-plugin-promise
 
 module.exports = {
+  // Promise Rules
+
   'promise/always-catch': 'off',
   // Ensure that each time a then() is applied to a promise, a catch() is applied as well. Exceptions are made if you are returning that promise.
   'promise/catch-or-return': 'warn',
+  // Avoid wrapping values in Promise.resolve or Promise.reject when not needed
+  'promise/no-return-wrap': 'error',
   // Ensure that inside a then() you make sure to return a new promise or value.
   'promise/always-return': 'error',
   // Enforce standard parameter names for Promise constructors.
   'promise/param-names': 'warn',
   // Ensure that Promise is included fresh in each file instead of relying on the existence of a native promise implementation.
   'promise/no-native': 'off',
+  // Avoid nested .then() or .catch() statements
+  'promise/no-nesting': 'warn',
+  // Avoid using promises inside of callbacks
+  'promise/no-promise-in-callback': 'warn',
+  // Avoid calling cb() inside of a then() (use nodeify] instead)
+  'promise/no-callback-in-promise': 'warn',
+   // Avoid creating new promises outside of utility libs (use pify instead)
+  'promise/avoid-new': 'warn',
+
+
+  // Async/Await Rules
+
+  // Prefer await to then() for reading Promise values
```

Comment:
Are we using the babel compiler for this or something else? Their implementation is not what I'd call light weight.

ex:
```javascript
async function doTheThing() {
  const whatever = await fetch('google.com');
  
  return whatever;
}
```

transforms to:

```javascript

var doTheThing = function () {
  var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
    var whatever;
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return fetch('google.com');

          case 2:
            whatever = _context.sent;
            return _context.abrupt('return', whatever);

          case 4:
          case 'end':
            return _context.stop();
        }
      }
    }, _callee, this);
  }));

  return function doTheThing() {
    return _ref.apply(this, arguments);
  };
}();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step("next", value); }, function (err) { step("throw", err); }); } } return step("next"); }); }; }
```
[source](http://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Ces2015-loose%2Ces2016%2Ces2017%2Clatest%2Creact%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3&experimental=false&loose=false&spec=false&code=async%20function%20doTheThing()%20%7B%0A%20%20const%20whatever%20%3D%20await%20fetch('google.com')%3B%0A%20%20%0A%20%20return%20whatever%3B%0A%7D&playground=true)

It also creates a blocking loop with the `while(1)` which is kinda scary. It does look like there's a defined exit condition, but it's hard to say from the generated code how many iterations it'll block on.